pkg/reconciler/namespace/controller.go

```diff
@@ -98,7 +101,17 @@ func NewController(
 		Handler: cache.ResourceEventHandlerFuncs{
 			AddFunc:    func(obj interface{}) { c.enqueueNamespace(obj) },
 			UpdateFunc: func(_, obj interface{}) { c.enqueueNamespace(obj) },
-			DeleteFunc: nil, // Nothing to do.
+			DeleteFunc: func(obj interface{}) {
+				// Ensure the scheduling cache entry for the namespace is removed
+				key, err := cache.MetaNamespaceKeyFunc(obj)
```

Comment:
Safer to use DeletionHandlingMetaNamespaceKeyFunc here

Reply:
Done.